include/dca/function/function.hpp

```diff
@@ -285,9 +294,16 @@ function<scalartype, domain>::function(const std::string& name)
       size_sbdm(dmn.get_leaf_domain_sizes()),
       step_sbdm(dmn.get_leaf_domain_steps()),
       fnc_values(nullptr) {
-  fnc_values = new scalartype[Nb_elements];
-  for (int linind = 0; linind < Nb_elements; ++linind)
-    setToZero(fnc_values[linind]);
+  if(name.substr(0, 2) == "G4" && distributed_g4_enabled)
```

Comment:
This switch on runtime type "name" is code smell, most of the chances we have of the compiler enforcing correct code when dealing with a distributed function is lost.  We should find a better way to do this. I'd prefer to have a template class specialization for distributed functions, but I think this is a good bootstrapping step.

Reply:
Good point! I will make it compile-time and let me research it on how to do it more elegantly.

Reply:
And remember we're c++17 now so it may be easier than it would have been.